third_party/terraform/resources/resource_google_service_account.go

```diff
@@ -131,29 +137,44 @@ func resourceGoogleServiceAccountDelete(d *schema.ResourceData, meta interface{}
 
 func resourceGoogleServiceAccountUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
+
 	sa, err := config.clientIAM.Projects.ServiceAccounts.Get(d.Id()).Do()
 	if err != nil {
 		return fmt.Errorf("Error retrieving service account %q: %s", d.Id(), err)
 	}
+
+	serviceAccount := &iam.ServiceAccount{
+		Etag: sa.Etag,
+	}
 	updateMask := make([]string, 0)
+
 	if d.HasChange("description") {
+		if v, ok := d.GetOk("description"); ok {
```

Comment:
The GCP client libraries don't send empty values unless you ask it to via `ForceSendFields`, so you don't actually need to do the `d.GetOk` check here.

Reply:
I'm also not sure why these needed to be moved out of where they were before. What exactly does doing this fix?